src/components/Calendar/Calendar.tsx

```diff
@@ -89,21 +161,27 @@ const Calendar: React.FC<CalendarProps> = ({
     initialEvents,
     view = "dayGridMonth",
     editable = false,
-    handleDateClick,
     initialDate,
 }) => {
     const [eventClick, setEventClick] = useState<CalendarEvent | null>(null);
+    const calendarRef = useRef<FullCalendar>(null) as React.MutableRefObject<FullCalendar>;
 
     const handleEventClick = (info: EventClickArg): void => {
         const id = info.event.id;
         setEventClick(initialEvents.find((event) => event.id === id) ?? null);
     };
 
+    const handleDateClick = (info: DateClickArg): void => {
+        const calendarApi = calendarRef.current.getApi();
+        calendarApi.changeView("timeGridDay", info.dateStr);
+    };
+
     return (
         <>
             <EventModal eventClick={eventClick} setEventClick={setEventClick} />
             <CalendarStyling>
                 <FullCalendar
+                    ref={calendarRef}
```

Comment:
Where do we use this useRef? I very much might just have missed a bit of the code somewhere but I can't see why we need it.

Reply:
It is used in calendarRef.current.getApi()